narwhal/types/src/primary.rs

```diff
@@ -30,9 +31,61 @@ use std::{
 /// The round number.
 pub type Round = u64;
 
+/// The epoch UNIX timestamp in milliseconds
+pub type TimestampMs = u64;
```

Comment:
Curious about your thoughts on why you decided to go with a new time-related type here instead of using `SystemTime` or something like `chrono::DateTime`? I am very wary of reinventing the wheel on time-related functions when they are notoriously tricky.

Reply:
So the main reason was the serialisation/deserialisation of the timestamp and avoid having the challenges that come with it (with the most "friendly" being the `chrono::DateTime` you referred to). I found that just relying on miliseconds (the `TimestampMs` is nothing more that just a `u64` in the end of the day) simplify things quite a lot. 

Another reason was the derives that some structs require (ex for `Batch` the `Arbitrary` https://github.com/MystenLabs/sui/blob/67b789e559d3c074ffcfb7848ee013afd5dc46fd/narwhal/types/src/primary.rs#L33 , or the `Default` one) which are not supported by the `SystemTime` or `chrono::DateTime` structs and again have to do some special handling.